File: env/load_env.R
# load_env.R - Load AWS environment variables for R

load_env <- function() {
  # Get the directory where this script is located
  script_dir <- dirname(rstudioapi::getActiveDocumentContext()$path)
  
  # If running outside RStudio, use current working directory
  if (identical(script_dir, "")) {
    script_dir <- getwd()
  }
  
  # Check for .env or .env.example
  env_file <- file.path(script_dir, ".env")
  env_example <- file.path(script_dir, ".env.example")
  
  if (file.exists(env_file)) {
    target_file <- env_file
    source_name <- ".env"
  } else if (file.exists(env_example)) {
    target_file <- env_example
    source_name <- ".env.example (using defaults)"
  } else {
    stop(paste("✗ Error: No .env or .env.example file found in", script_dir))
  }
  
  # Parse and load environment variables
  tryCatch({
    lines <- readLines(target_file)
    
    for (line in lines) {
      line <- trimws(line)
      
      # Skip empty lines and comments
      if (nchar(line) == 0 || startsWith(line, "#")) {
        next
      }
      
      # Parse KEY=VALUE
      if (grepl("=", line)) {
        parts <- strsplit(line, "=", fixed = TRUE)[[1]]
        if (length(parts) == 2) {
          key <- trimws(parts[1])
          value <- trimws(parts[2])
          Sys.setenv(key = value)
        }
      }
    }
    
    cat("✓ Loaded environment variables from", source_name, "\n")
    
    # Verify critical variables
    required_vars <- c('AWS_ACCOUNT_ID_PRIMARY', 'AWS_ACCOUNT_ID_LAMBDA')
    missing <- required_vars[sapply(required_vars, function(x) Sys.getenv(x) == "")]
    
    if (length(missing) > 0) {
      stop(paste("✗ Error: Missing required environment variables:", paste(missing, collapse = ", ")))
    }
    
    cat("✓ Environment loaded successfully\n")
    cat("  Primary Account:", Sys.getenv("AWS_ACCOUNT_ID_PRIMARY"), "\n")
    cat("  Lambda Account:", Sys.getenv("AWS_ACCOUNT_ID_LAMBDA"), "\n")
    cat("  Region (Primary):", Sys.getenv("AWS_REGION_PRIMARY"), "\n")
    cat("  Region (Lambda):", Sys.getenv("AWS_REGION_LAMBDA"), "\n")
    
  }, error = function(e) {
    stop(paste("✗ Error loading environment variables:", e$message))
  })
}

# Auto-run when script is sourced
if (!interactive()) {
  load_env()
}

File: env/load_env.py
import os
import sys
from pathlib import Path

def load_env():
    """Load AWS environment variables from .env or .env.example"""
    
    script_dir = Path(__file__).parent
    env_file = script_dir / ".env"
    env_example = script_dir / ".env.example"
    
    # Choose which file to load
    if env_file.exists():
        target_file = env_file
        source = ".env"
    elif env_example.exists():
        target_file = env_example
        source = ".env.example (using defaults)"
    else:
        print(f"✗ Error: No .env or .env.example file found in {script_dir}")
        sys.exit(1)
    
    # Parse and load environment variables
    try:
        with open(target_file, 'r') as f:
            for line in f:
                line = line.strip()
                # Skip empty lines and comments
                if not line or line.startswith('#'):
                    continue
                
                # Parse KEY=VALUE
                if '=' in line:
                    key, value = line.split('=', 1)
                    os.environ[key.strip()] = value.strip()
        
        print(f"✓ Loaded environment variables from {source}")
        
        # Verify critical variables
        required_vars = ['AWS_ACCOUNT_ID_PRIMARY', 'AWS_ACCOUNT_ID_LAMBDA']
        missing = [var for var in required_vars if var not in os.environ]
        
        if missing:
            print(f"✗ Error: Missing required environment variables: {', '.join(missing)}")
            sys.exit(1)
        
        print("✓ Environment loaded successfully")
        print(f"  Primary Account: {os.environ.get('AWS_ACCOUNT_ID_PRIMARY')}")
        print(f"  Lambda Account: {os.environ.get('AWS_ACCOUNT_ID_LAMBDA')}")
        print(f"  Region (Primary): {os.environ.get('AWS_REGION_PRIMARY')}")
        print(f"  Region (Lambda): {os.environ.get('AWS_REGION_LAMBDA')}")
        
    except Exception as e:
        print(f"✗ Error loading environment variables: {e}")
        sys.exit(1)


if __name__ == "__main__":
    load_env()
